docs/gallery/general/plot_resources.py

```diff
@@ -0,0 +1,396 @@
+  :py:class:`~pynwb.resources.Object`
+* :py:class:`~pynwb.resources.ObjectKeyTable` where each row describes an
+  :py:class:`~pynwb.resources.ObjectKey` pair identifying which keys
+  are used by which objects.
+
+The :py:class:`~pynwb.resources.HERD` class then provides
+convenience functions to simplify interaction with these tables, allowing users
+to treat :py:class:`~pynwb.resources.HERD` as a single large table as
+much as possible.
+
+Rules to HERD
+---------------------------
+When using the :py:class:`~pynwb.resources.HERD` class, there
+are rules to how users store information in the interlinked tables.
+
+1. Multiple :py:class:`~pynwb.resources.Key` objects can have the same name.
+   They are disambiguated by the :py:class:`~pynwb.resources.Object` associated
+   with each, meaning we may have keys with the same name in different objects, but for a particular object
+   all keys must be unique.
+2. In order to query specific records, the :py:class:`~pynwb.resources.HERD` class
+   uses '(file, object_id, relative_path, field, key)' as the unique identifier.
+3. :py:class:`~pynwb.resources.Object` can have multiple :py:class:`~pynwb.resources.Key`
+   objects.
+4. Multiple :py:class:`~pynwb.resources.Object` objects can use the same :py:class:`~pynwb.resources.Key`.
+5. Do not use the private methods to add into the :py:class:`~pynwb.resources.KeyTable`,
+   :py:class:`~pynwb.resources.FileTable`, :py:class:`~pynwb.resources.EntityTable`,
+   :py:class:`~pynwb.resources.ObjectTable`, :py:class:`~pynwb.resources.ObjectKeyTable`,
+   :py:class:`~pynwb.resources.EntityKeyTable` individually.
+6. URIs are optional, but highly recommended. If not known, an empty string may be used.
+7. An entity ID should be the unique string identifying the entity in the given resource.
+   This may or may not include a string representing the resource and a colon.
+   Use the format provided by the resource. For example, Identifiers.org uses the ID ``ncbigene:22353``
+   but the NCBI Gene uses the ID ``22353`` for the same term.
+8. In a majority of cases, :py:class:`~pynwb.resources.Object` objects will have an empty string
+   for 'field'. The :py:class:`~pynwb.resources.HERD` class supports compound data_types.
+   In that case, 'field' would be the field of the compound data_type that has an external reference.
+9. In some cases, the attribute that needs an external reference is not a object with a 'data_type'.
+   The user must then use the nearest object that has a data type to be used as the parent object. When
+   adding an external resource for an object with a data type, users should not provide an attribute.
+   When adding an external resource for an attribute of an object, users need to provide
+   the name of the attribute.
+10. The user must provide a :py:class:`~pynwb.resources.File` or an :py:class:`~pynwb.resources.Object` that
+    has :py:class:`~pynwb.resources.File` along the parent hierarchy.
+"""
```

Comment:
I think this is too much low-level detail. This is information that we should point to the tutorial in HDMF for. In the PyNWB tutorial we should focus on NWB-specific examples and walk through the typical use with NWB. 

As such, I would suggest a structure that focuses more directly on the different use-cases of NWB Users:
1) Annotate data while creating a new `NWBFile`:
   * Show and example where `Species` is wrapped with a TermSetWrapper works
   * Show how wrapping a whole `VectorData` column with TermSetWrapper works
   * Show how wrapping with TermSetWrapper allows one to: a) validate data and b) automatically create at the end a HERD file
   * Show and example where add_ref is being used to add a custom reference (e.g., for experimenter)
2) Show how annotating an existing `NWBFile` works (here I think we can use the DANDIset example)
3) Show how reading and querying a HERD file works

Reply:
Yes it's part of the PR TODO items to do pynwb examples and not hdmf